doc/pages/command-parsing.asciidoc

```diff
@@ -9,6 +9,14 @@ prompt, are parsed according to the following rules:
 
 - Words (command names and parameters) are delimited by whitespaces.
 
+- <<expansions#, Expansions>> are parsed and replaced before the execution of
+  the commands.
+
+== Comments parsing
+
+Commands are commented out after `#` until end of line (not `;`).
+`#` takes effect _after_ the expansions are parsed.
```

Comment:
That sounds incorrect to me, `decl str foo 'foo # bar'; echo %opt{foo}` does echo `foo # bar`, or maybe you meant something different with "after the expansions are parsed"

Reply:
I meant that it's only executed, if it is, only after the expansions are parsed. But reading this again and after your observation, it's clearly poorly worded.

How about:
`The parsing of strings and expansions happens before `#` is evaluated as commenting out the rest of the line, hence its meaning might change according to command-parsing.`